xdsl/transforms/lower_mpi.py

```diff
@@ -211,7 +214,7 @@ def _emit_memref_counts(
         if not all(dim.value.data >= 0 for dim in ssa_val.typ.shape.data):
             raise RuntimeError("MPI lowering does not support unknown-size memrefs!")
 
-        size = sum(dim.value.data for dim in ssa_val.typ.shape.data)
+        size = prod(dim.value.data for dim in ssa_val.typ.shape.data)
```

Comment:
seems like a useful helper to add on the type, potentially?

Reply:
Absolutely!